rsocket-core/src/main/java/io/rsocket/exceptions/RSocketException.java

```diff
@@ -16,17 +16,42 @@
 
 package io.rsocket.exceptions;
 
+import java.util.Objects;
+
+/** The root of the RSocket exception hierarchy. */
 public abstract class RSocketException extends RuntimeException {
 
   private static final long serialVersionUID = 2912815394105575423L;
 
+  /**
+   * Constructs a new exception with the specified message.
+   *
+   * @param message the message
+   * @throws NullPointerException if {@code message} is {@code null}
+   */
   public RSocketException(String message) {
-    super(message);
+    super(Objects.requireNonNull(message, "message must not be null"));
   }
 
+  /**
+   * Constructs a new exception with the specified message and cause.
+   *
+   * @param message the message
+   * @param cause the cause of this exception
+   * @throws NullPointerException if {@code message} or {@code cause} is {@code null}
+   */
   public RSocketException(String message, Throwable cause) {
-    super(message, cause);
+    super(
+        Objects.requireNonNull(message, "message must not be null"),
+        Objects.requireNonNull(cause, "cause must not be null"));
```

Comment:
nit: this seems overly strict given that cause in RuntimeException is nullable.  As it's in error handling code, being super strict here is likely to just translate some business errors into rarely seen NPE.

Reply:
I’d wondered about this as I did it. Specifically, this was codifying [existing intent](https://github.com/rsocket/rsocket-java/blob/1.0.x/rsocket-core/src/main/java/io/rsocket/exceptions/package-info.java#L17). I’m happy to remove if you think that it the intent was inaccurate.

Reply:
My concern wouldn't be with the ApplicationException created by users of the library ever passing in something like

`new ApplicationException("failed reading from DB", e.getCause())`

I'd weakly vote for removing, but wouldn't fight to the death over it.

Reply:
@yschimke I'd agree with you, and since I haven't seen any objections here I'll update.